### PR TITLE
CORE-4817 Create in-memory fake CpiInfoReadService

### PR DESCRIPTION
**Pull request by abellagonzalo**
Fake implementation of the CpiInfoReadService.

There are a few things that should be improved:

* Remove duplicated classes:
  * `net.corda.libs.packaging.Cpi.Identifier` and `net.corda.libs.packaging.core.CpiIdentifier`
  * `net.corda.libs.packaging.Cpi.Metadata` and `net.corda.libs.packaging.core.CpiMetadata`
* Load CPI Info from yaml file
* Load CPI Info from file
* Create a BaseFakeService class with common functionality?
* Improve Test Catalogues to return proper initialise data and extract them into independent modules so they can be reused across multiple projects.

### CODE REVIEW

**Review by lankydan — COMMENTED** [review]
Owen can probably say better but it looks ok to me.

**Review by lankydan — DISMISSED** [review]

File: testing/cpi-info-read-service-fake/src/test/kotlin/net/corda/cpiinfo/read/fake/TestCatalogue.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import net.corda.libs.packaging.CordappManifest
+import net.corda.libs.packaging.Cpk
+import net.corda.libs.packaging.ManifestCordappInfo
+import net.corda.libs.packaging.core.CpiIdentifier
+import net.corda.libs.packaging.core.CpiMetadata
+import net.corda.libs.packaging.core.CpkIdentifier
+import net.corda.libs.packaging.core.CpkMetadata
+import net.corda.v5.crypto.SecureHash
+
+object TestCatalogue {
```
Comment (driessamyn) [inline testing/cpi-info-read-service-fake/src/test/kotlin/net/corda/cpiinfo/read/fake/TestCatalogue.kt:14]: @josephzunigadaly FYI

**Review by driessamyn — DISMISSED** [review]
LGTM

**Review by driessamyn — APPROVED** [review]
LGTM

**Pull request merged**